src/storage/store/column.cpp

```diff
@@ -43,21 +43,17 @@ struct ReadInternalIDValuesToVector {
 
 struct WriteInternalIDValuesToPage {
     WriteInternalIDValuesToPage() : compressedWriter{LogicalType(LogicalTypeID::INTERNAL_ID)} {}
-    void operator()(uint8_t* frame, uint16_t posInFrame, const uint8_t* data, uint32_t dataOffset,
-        offset_t numValues, const CompressionMetadata& metadata) {
-        auto internalIDValues = reinterpret_cast<const internalID_t*>(data);
-
-        for (auto i = 0u; i < numValues; i++) {
-            compressedWriter(frame, posInFrame,
-                reinterpret_cast<const uint8_t*>(&internalIDValues[dataOffset + i].offset),
-                0 /*dataOffset*/, 1 /*numValues*/, metadata);
-        }
+    void operator()(uint8_t* /*frame*/, uint16_t /*posInFrame*/, const uint8_t* /*data*/,
+        uint32_t /*dataOffset*/, offset_t /*numValues*/, const CompressionMetadata& /*metadata*/) {
+        KU_ASSERT(false);
     }
     void operator()(uint8_t* frame, uint16_t posInFrame, ValueVector* vector,
         uint32_t offsetInVector, const CompressionMetadata& metadata) {
         KU_ASSERT(vector->dataType.getPhysicalType() == PhysicalTypeID::INTERNAL_ID);
-        compressedWriter(
-            frame, posInFrame, vector->getData(), offsetInVector, 1 /*numValues*/, metadata);
+        auto internalIDValues = reinterpret_cast<const internalID_t*>(vector->getData());
+        compressedWriter(frame, posInFrame,
+            reinterpret_cast<const uint8_t*>(&internalIDValues[offsetInVector].offset),
```

Comment:
Maybe `vector->getValue(offsetInVector)` would be better here, if only to delegate the cast to ValueVector.